### PR TITLE
[DOCS] Reformat query string query

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the `query_string` query to use the new query format.

This creates separate sections for the example request, parameters, and notes.

This is part of #40977, an effort to standardize documentation for query types.

### Preview
http://elasticsearch_45296.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-query-string-query.html

### CODE REVIEW

File: docs/reference/query-dsl/match-query.asciidoc [hunk 1]
```diff
@@ -157,15 +157,3 @@ The example above creates a boolean query:
 that matches documents with the term `ny` or the conjunction `new AND york`.
 By default the parameter `auto_generate_synonyms_phrase_query` is set to `true`.
 
-
-.Comparison to query_string / field
-**************************************************
-
```
Comment (jrodewig) [inline docs/reference/query-dsl/match-query.asciidoc]: Reworded and moved to `query-string-query.asciidoc`:
https://github.com/elastic/elasticsearch/pull/45296/files#diff-d1da0885c6d0812eedd305523174ba5cR19

File: docs/reference/query-dsl/query-string-query.asciidoc [hunk 2]
```diff
@@ -4,163 +4,251 @@
...
+syntax, use the <<query-dsl-simple-query-string-query,`simple_query_string`>>
+query, which is less strict.
+====
+ 
+
+[[query-string-query-ex-request]]
+==== Example request
+
+When running the following search, the `query_string` query splits `(new york
+city) OR (big apple)` into two terms: `new york city` and `big apple`. The
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/query-string-query.asciidoc]: > into two terms

`term` has another special meaning in ES/Lucene; a term is an output from an analyzer.  Let call these "into two parts" or something like this.

File: docs/reference/query-dsl/query-string-query.asciidoc [hunk 3]
```diff
@@ -4,163 +4,251 @@
...
+`content` field's analyzer then independently converts each term into tokens
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/query-string-query.asciidoc]: "term" - similarly here, it is better to call it something else.

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jrodewig. Nice work!

**Comment by jrodewig** [comment]
Thanks for your review @mayya-sharipova!

**Pull request merged**